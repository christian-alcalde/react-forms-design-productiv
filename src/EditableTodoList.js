import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({ todos, update, remove }) {
  
  function renderTodoList() {
    return todos.map((todo) => (
      <li key={todo.id}>
        <EditableTodo todo={todo} update={update} remove={remove} />
      </li>
    ));
  }

  return (
    <div>
      <ul>{renderTodoList()}</ul>
    </div>
  )
}

export default EditableTodoList;
