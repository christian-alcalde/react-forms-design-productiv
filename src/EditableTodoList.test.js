import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";


const testTodos = [{
  id: "1",
  title: "test",
  description: "testDescription",
  priority: 1,
}, {
  id: "2",
  title: "test2",
  description: "testDescription2",
  priority: 2,
}];


it("renders without crashing", function () {
  render(<EditableTodoList todos={testTodos} update={jest.fn} remove={jest.fn} />);
});

it("matches snapshot", function () {
  const { container } = render(
    <EditableTodoList todos={testTodos} update={jest.fn} remove={jest.fn} />
  );
  expect(container).toMatchSnapshot();
});

it("two items rendered in list", function () {
  const { container } = render(
    <EditableTodoList todos={testTodos} update={jest.fn} remove={jest.fn} />
  );

  expect(container).toContainHTML("test")
  expect(container).toContainHTML("test2")
});
