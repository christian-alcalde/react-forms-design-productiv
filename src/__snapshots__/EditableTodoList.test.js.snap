// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <ul>
      <li
        data-testid="1"
      >
        <div
          class="EditableTodo"
        >
          <div
            class="mb-3"
          >
            <div
              class="float-end text-sm-end"
            >
              <button
                class="EditableTodo-toggle btn-link btn btn-sm"
              >
                Edit
              </button>
              <button
                class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
              >
                Del
              </button>
            </div>
            <div
              class="Todo"
            >
              <div>
                <b>
                  test
                </b>
                 
                <small>
                  (Priority:
                  1
                  )
                </small>
              </div>
              <div>
                <small>
                  testDescription
                </small>
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        data-testid="2"
      >
        <div
          class="EditableTodo"
        >
          <div
            class="mb-3"
          >
            <div
              class="float-end text-sm-end"
            >
              <button
                class="EditableTodo-toggle btn-link btn btn-sm"
              >
                Edit
              </button>
              <button
                class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
              >
                Del
              </button>
            </div>
            <div
              class="Todo"
            >
              <div>
                <b>
                  test2
                </b>
                 
                <small>
                  (Priority:
                  2
                  )
                </small>
              </div>
              <div>
                <small>
                  testDescription2
                </small>
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
